src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -396,6 +395,10 @@ private HttpClient CreateHttpClient()
                 }
             }
 
+            // Add the HttpOptions handler, this will apply any modifications to the http request
+            // (like applying headers, or the access token and adding the user agent).
+            httpMessageHandler = new HttpOptionsMessageHandler(httpMessageHandler, _httpOptions);
```

Comment:
Only wrap the handler if `_httpOptions` isn't null? Then you can remove null options logic from the handler.

Reply:
Sounds good.

Reply:
Although you always want to set `request.Headers.UserAgent.Add(Constants.UserAgentHeader);` even if options are null.

I think you should rename the handler name if it isn't just for setting `HttpOptions`.